rex/renewable_resource.py

```diff
@@ -403,6 +412,87 @@ def get_nearest_h(h, heights):
 
         return nearest_h, extrapolate
 
+    @staticmethod
+    def monin_obukhov_extrapolation(ts_1, h_1, z0, L, h):
+        """
+        Monin-Obukhov extrapolation
+
+        Parameters
+        ----------
+         ts_1 : ndarray
+            Time-series array at height h_1
+        h_1 : int | float
+            Height corresponding to time-seris ts_1
+        z0: int | float | ndarray
+            Roughness length
+        L : ndarray
+            time-series of Obukhov length (m; measure of stability)
+        h : int | float
+            Desired height
+
+        Returns
+        -------
+        ndarray
+            new wind speed from MO extrapolation.
+        """
+        # Non dimensional stability parameter at h
+        zeta = WindResource.stability_function(h / L)
+        # Non dimensional stability parameter at z0
+        zeta_0 = WindResource.stability_function(z0 / L)
+        # Non dimensional stability parameter at h_1
+        zeta_1 = WindResource.stability_function(h_1 / L)
+
+        # Logarithmic extrapolation equation
+        out = (ts_1 * (np.log(h / z0) - zeta + zeta_0)
+               / (np.log(h_1 / z0) - zeta_1 + zeta_0))
+
+        return out
+
+    @staticmethod
+    def stability_function(zeta):
+        """
+        Calculate stability function depending on sign of L
+        (negative is unstable, positive is stable)
+
+        Parameters
+        ----------
+        zeta : ndarray
+            Normalized length
+
+        Returns
+        -------
+        numpy.ndarray
+            stability measurements.
+        """
+        stab_fun = np.zeros(len(zeta))
+        zeta = zeta.astype(float)
+
+        # Unstable conditions
+        x = (np.power(1 - 16 * zeta[zeta < 0], 0.25))
+        paulson_func = (np.pi / 2 - 2 * np.arctan(x)
```

Comment:
Going to trust you on this one....

Reply:
LOL, just stole this from [RVT](https://github.nrel.gov/wind-toolkit/RVT), mike optis coded it up, so I'm trusting him...